Write up documentation on posthog-dotnet (#10589)
PostHog/posthog.com: contents/docs/integrate/feature-flags-code/_snippets/feature-flags-code-dotnet.mdx

There are 2 steps to implement feature flags in .NET:

### Step 1: Evaluate the feature flag value

#### Boolean feature flags

```csharp
var isMyFlagEnabled = await posthog.IsFeatureEnabledAsync(
    "flag-key",
    "distinct_id_of_your_user"
);
if (isMyFlagEnabled.GetValueOrDefault())
{
    // Feature is enabled
}
else
{
    // Feature is disabled    
}
```

> **Note:** The `IsFeatureEnabledAsync` method returns a nullable boolean. If the flag is not found or evaluating it is inconclusive, it returns `null`.

Reviewer: ```suggestion
> **Note:** The `IsFeatureEnabledAsync` method returns a nullable boolean. If the flag is not found or evaluating it is inconclusive, it returns `null`.
```

#### Multivariate feature flags

```csharp
var flag = await posthog.GetFeatureFlagAsync(
    "flag-key",
    "distinct_id_of_your_user"
);

// replace "variant-key" with the key of your variant
if (flag is { VariantKey: "variant-key"} ) {
    // Do something differently for this user
    // Optional: fetch the payload
    var matchedPayload = flag.Payload;
}
```

The `GetFeatureFlagAsync` method returns a nullable `FeatureFlag` object. If the flag is not found or evaluating it is inconclusive, it returns `null`. However, there is an implicit conversion to bool to make comparisons easier.

```csharp
if (await posthog.GetFeatureFlagAsync(
    "flag-key",
    "distinct_id_of_your_user")
)
{
    // Do something differently for this user
}
```

import IncludePropertyInEvents from "./include-feature-flag-property-in-backend-events.mdx" 

<IncludePropertyInEvents />

There are two methods you can use to include feature flag information in your events:

#### Method 1: Include the `$feature/feature_flag_name` property

>  In the event properties, include `$feature/feature_flag_name: variant_key`:

Reviewer: ```suggestion
In the event properties, include `$feature/feature_flag_name: variant_key`:
```

    "event_name",
    properties: new() {
        // replace feature-flag-key with your flag key.
        // Replace 'variant-key' with the key of your variant
        ["$feature/feature-flag-key"] = "variant-key"
    }
);
```

#### Method 2: Set `send_feature_flags` to `true`

import DotNetSetSendFeatureFlagsTrue from "./feature-flags-code-dotnet-set-send-feature-flags-to-true.mdx" 

<DotNetSetSendFeatureFlagsTrue />

### Fetching all flags for a user

You can fetch all flag values for a single user by calling `GetAllFeatureFlagsAsync()`.

This is useful when you need to fetch multiple flag values and don't want to make multiple requests.

```csharp  
var flags = await posthog.GetAllFeatureFlagsAsync(
    "distinct_id_of_your_user"
);
```

### Sending `$feature_flag_called` events

Capturing `$feature_flag_called` events enable PostHog to know when a flag was accessed by a user and thus provide [analytics and insights](/docs/product-analytics/insights) on the flag. By default, we send a these event when:

1. You call `posthog.GetFeatureFlagAsync()` or `posthog.IsFeatureEnabledAsync()`, AND 
2. It's a new user, or the value of the flag has changed. 

> > *Note:* Tracking whether it's a new user or if a flag value has changed happens in a local cache. This means that if you reinitialize the PostHog client, the cache resets as well – causing `$feature_flag_called` events to be sent again when calling `GetFeatureFlagAsync` or `IsFeatureEnabledAsync`. PostHog is built to handle this, and so duplicate `$feature_flag_called` events won't affect your analytics.

Reviewer: ```suggestion
> **Note:** Tracking whether it's a new user or if a flag value has changed happens in a local cache. This means that if you reinitialize the PostHog client, the cache resets as well – causing `$feature_flag_called` events to be sent again when calling `GetFeatureFlagAsync` or `IsFeatureEnabledAsync`. PostHog is built to handle this, and so duplicate `$feature_flag_called` events won't affect your analytics.
```

> You can disable automatically the additional request to capture `$feature_flag_called` events. For example, when you don't need the analytics, or it's being called at such a high volume that sending events slows things down.

Reviewer: ```suggestion
You can disable automatic capture of these events and manually capture `$feature_flag_called` events instead. This is helpful when you don't need the analytics, or it's being called at such a high volume that sending events slows things down.
```

To disable it, set the `sendFeatureFlagsEvent` option in your function call, like so:

```csharp
var isMyFlagEnabled = await posthog.IsFeatureEnabledAsync(
    "flag-key", 
    "distinct_id_of_your_user",
    options: new FeatureFlagOptions
    {
        SendFeatureFlagEvents = true
    }
);

> // will not send `$feature_flag_called` events

Reviewer: should this be false?
